src/message.cpp

```diff
@@ -222,9 +222,12 @@ void warn_simple(const char *file,int line,const char *text)
 
 void warn_undoc(const char *file,int line,const char *fmt, ...)
 {
+  static bool warnIfUndocumented = Config_getBool(WARN_IF_UNDOCUMENTED);
+  static bool extractAll = Config_getBool(EXTRACT_ALL);
+  static bool warnExtractAll = Config_getBool(WARN_EXTRACT_ALL);
   va_list args;
   va_start(args, fmt);
-  do_warn(Config_getBool(WARN_IF_UNDOCUMENTED), file, line, warning_str, fmt, args);
+  do_warn(warnIfUndocumented || (extractAll && warnExtractAll), file, line, warning_str, fmt, args);
```

Comment:
This logic seems wrong to me. If extractAll and warnExtractAll are both enabled, then the warnIfUndocumented setting is ignored.

Reply:
I think the logic is correct and is intentionally. When `EXTRACT_ALL` and `WARN_EXTRACT_ALL` are set we want to have all warnings, irrespective of the setting of `WARN_IF_UNDOCUMENTED`.